echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/DockerEventMonitor.java

```diff
@@ -0,0 +1,113 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pipelinetriggers.monitor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.echo.model.Event;
+import com.netflix.spinnaker.echo.model.Pipeline;
+import com.netflix.spinnaker.echo.model.Trigger;
+import com.netflix.spinnaker.echo.model.trigger.DockerEvent;
+import com.netflix.spinnaker.echo.model.trigger.TriggerEvent;
+import com.netflix.spinnaker.echo.pipelinetriggers.PipelineCache;
+import lombok.NonNull;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import rx.Observable;
+import rx.functions.Action1;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+@Component
+public class DockerEventMonitor extends TriggerMonitor {
+
+  public static final String TRIGGER_TYPE = "docker";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final PipelineCache pipelineCache;
+
+  @Autowired
+  public DockerEventMonitor(@NonNull PipelineCache pipelineCache,
+                @NonNull Action1<Pipeline> subscriber,
+                @NonNull Registry registry) {
+    super(subscriber, registry);
+    this.pipelineCache = pipelineCache;
+  }
+
+  @Override
+  public void processEvent(Event event) {
+    if (!event.getDetails().getType().equalsIgnoreCase(DockerEvent.TYPE)) {
+      return;
+    }
+
+    DockerEvent dockerEvent = objectMapper.convertValue(event, DockerEvent.class);
+    Observable.just(dockerEvent)
+      .doOnNext(this::onEchoResponse)
+      .subscribe(triggerEachMatchFrom(pipelineCache.getPipelines()));
+  }
+
+  @Override
+  protected boolean isSuccessfulTriggerEvent(final TriggerEvent event) {
+    DockerEvent dockerEvent = (DockerEvent) event;
+    String digest = dockerEvent.getContent().getDigest();
+    return digest != null && !digest.isEmpty();
+  }
+
+  @Override
+  protected Function<Trigger, Pipeline> buildTrigger(Pipeline pipeline, TriggerEvent event) {
+    DockerEvent dockerEvent = (DockerEvent) event;
+    return trigger -> pipeline.withTrigger(trigger.atDigest(dockerEvent.getContent().getDigest()));
+  }
+
+  @Override
+  protected boolean isValidTrigger(final Trigger trigger) {
+    return trigger.isEnabled() &&
+      (
+        (TRIGGER_TYPE.equals(trigger.getType()) &&
+          trigger.getRegistry() != null &&
+          trigger.getRepository() != null &&
+          trigger.getTag() != null)
+      );
+  }
+
+  @Override
+  protected Predicate<Trigger> matchTriggerFor(final TriggerEvent event) {
+    DockerEvent dockerEvent = (DockerEvent) event;
+    String registry = dockerEvent.getContent().getRegistry();
+    String repository = dockerEvent.getContent().getRepository();
+    String tag = dockerEvent.getContent().getTag();
+    return trigger ->  trigger.getType().equals(TRIGGER_TYPE) &&
```

Comment:
Not sure what the typical style is here, but there are two spaces after the `->` operator.